Add unify-opening

DamienCassou opened the pull request:
summary: Emacs package to make everything use the same mechanism to open
         files
repository: https://github.com/DamienCassou/unify-opening
association with package: author

milkypostman commented:
What exactly does this do? There is not much in terms of code there but I don't really understand the behavior here. Docs could use a bit more explanation for someone who is outside of mu4e and org.

purcell commented:
And with-eval-after-load is very new to Emacs. Perhaps 24.4?

purcell commented:
Also the use of the new advice functions requires an Emacs version dependency.

DamienCassou commented:
@milkypostman I've just added some comments in the README and Commentary section of the `.el` file.
@purcell I've just added a dependency to emacs 24.4. (I could have lowered the dependency by rewriting the code, but I will wait for someone to request that :-)).

Please merge

purcell commented:
Merci Damien -- merged!
